packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -232,109 +237,54 @@ class FlatTermSelector extends Component {
-			_x( '%s added', 'term' ),
-			singularName
-		);
-		const termRemovedLabel = sprintf(
-			/* translators: %s: term name. */
-			_x( '%s removed', 'term' ),
-			singularName
-		);
-		const removeTermLabel = sprintf(
-			/* translators: %s: term name. */
-			_x( 'Remove %s', 'term' ),
-			singularName
-		);
+	const newTermLabel = get(
+		taxonomy,
+		[ 'labels', 'add_new_item' ],
+		slug === 'post_tag' ? __( 'Add new tag' ) : __( 'Add new Term' )
+	);
+	const singularName = get(
+		taxonomy,
+		[ 'labels', 'singular_name' ],
+		slug === 'post_tag' ? __( 'Tag' ) : __( 'Term' )
+	);
+	const termAddedLabel = sprintf(
+		/* translators: %s: term name. */
+		_x( '%s added', 'term' ),
+		singularName
+	);
+	const termRemovedLabel = sprintf(
+		/* translators: %s: term name. */
+		_x( '%s removed', 'term' ),
+		singularName
+	);
+	const removeTermLabel = sprintf(
+		/* translators: %s: term name. */
+		_x( 'Remove %s', 'term' ),
+		singularName
+	);
 
-		return (
-			<>
-				<FormTokenField
-					value={ selectedTerms }
-					suggestions={ termNames }
-					onChange={ this.onChange }
-					onInputChange={ this.searchTerms }
-					maxSuggestions={ MAX_TERMS_SUGGESTIONS }
-					disabled={ loading }
-					label={ newTermLabel }
-					messages={ {
-						added: termAddedLabel,
-						removed: termRemovedLabel,
-						remove: removeTermLabel,
-					} }
-				/>
-				<MostUsedTerms
-					taxonomy={ taxonomy }
-					onSelect={ this.appendTerm }
-				/>
-			</>
-		);
-	}
+	return (
+		<>
+			<FormTokenField
+				value={ values }
+				suggestions={ suggestions }
+				onChange={ onChange }
+				onInputChange={ debouncedSearch }
+				maxSuggestions={ MAX_TERMS_SUGGESTIONS }
+				disabled={ isLoading }
+				label={ newTermLabel }
+				messages={ {
+					added: termAddedLabel,
+					removed: termRemovedLabel,
+					remove: removeTermLabel,
+				} }
+			/>
+			<MostUsedTerms taxonomy={ taxonomy } onSelect={ appendTerm } />
+		</>
+	);
 }
 
-export default compose(
-	withSelect( ( select, { slug } ) => {
-		const { getCurrentPost } = select( editorStore );
-		const { getTaxonomy } = select( coreStore );
-		const taxonomy = getTaxonomy( slug );
-		return {
-			hasCreateAction: taxonomy
-				? get(
-						getCurrentPost(),
-						[ '_links', 'wp:action-create-' + taxonomy.rest_base ],
-						false
-				  )
-				: false,
-			hasAssignAction: taxonomy
-				? get(
-						getCurrentPost(),
-						[ '_links', 'wp:action-assign-' + taxonomy.rest_base ],
-						false
-				  )
-				: false,
-			terms: taxonomy
-				? select( editorStore ).getEditedPostAttribute(
-						taxonomy.rest_base
-				  )
-				: [],
-			taxonomy,
-		};
-	} ),
-	withDispatch( ( dispatch ) => {
-		return {
-			onUpdateTerms( terms, restBase ) {
-				dispatch( editorStore ).editPost( { [ restBase ]: terms } );
-			},
-		};
-	} ),
-	withSpokenMessages,
-	withFilters( 'editor.PostTaxonomyType' )
-)( FlatTermSelector );
+export default withFilters( 'editor.PostTaxonomyType' )( FlatTermSelector );
```

Comment:
Similar to the other PR, if we want to land this, we may need to check plugins using that hook and warning them.